contribute/code.rst

```diff
@@ -0,0 +1,211 @@
+
+Rebase your changes during review, if possible
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It's common for a commit log on GitHub to go something like this:
+
+* Fix bug in phone number search
+* Add new API call for new phone number search
+* Add UI for new phone number search
+* Fix UI
+* Fix bug fix
+* Review changes
+* Fix API after comments from reviewers
+
+If you have to come back to this commit log in the future, you'll be confused in no time at all. What did the reviewers say? What was wrong with the UI? What was wrong with the bug fix?
+
+When you receive a request for changes that affects the first commit in your series, edit that commit instead of adding a new one. If doing this changes the information in your commit message, update that too.
+
+You will have to learn how to use Git to edit the series of commits in your change and upload your new series. You will generally edit your commit series with an interactive git rebase. Then you will force-push your new series over the old one or make a new branch and open a new changeset. These are not perfect solutions, but they are the best we have.
+
+Luckily, there are graphical tools that make this job easier. Unluckily, if you don't understand what they are doing on the Git command line, you might mess up and have to start over from your old series. Even seasoned Git pros mess up this process sometimes, don't worry. There is almost always a way back to your old series.
+
+Each commit must continue to build, run, and pass tests after your changes.
+
+Format your commit message correctly
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Git gurus prefer that you keep the first line of your commit message (the summary) to 50 characters or less. They also prefer that no line in the commit message goes over 72 characters. It's fine if you have to break the rules — some changes can't be summarized in 50 characters and some links are longer than 72.
+
+Changeset description checklist
+-------------------------------
+
+Ensure that your changeset answers the following questions in its description.
+
+If you've followed the guidelines for commit messages, you can probably copy the relevant information from the commit messages into your changeset description. GitHub and GitLab will do this for you in single-commit changesets. A description that says "this is complicated to explain, please read the commit messages" is also fine. We trust you to strike the balance.
+
```

Comment:
I think a note about adding screenshots would be good here, maybe:

If you did any changes regarding the user interface, provide screenshots showing the appearance  before and after.